### ISSUE TITLE
High memory usage - 9,6GiB

### ISSUE THREAD

**Issue by jkaberg** [ONLYOFFICE/Docker-DocumentServer#69]
Is 9,6GiB to be expected when running OnlyOffice? On avarage 1 user/day editing 1-2 documents...

```
# docker stats onlyoffice_1 --no-stream
CONTAINER           CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O           PIDS
onlyoffice_1   1.97%               9.642GiB / 47.16GiB   20.44%              30.9MB / 67.2MB     566MB / 784MB       177

```

**Comment by jkaberg** [comment]
Looking into this further reveals,

```
root@69ccdaf9dcd5:/# ps aux  | awk '{print $6/1024 " MB\t\t" $11}'  | sort -n
0 MB            COMMAND
0.09375 MB              /usr/lib/erlang/erts-5.10.4/bin/epmd
0.484375 MB             bash
0.835938 MB             /bin/sh
0.839844 MB             /bin/sh
0.867188 MB             inet_gethost
0.988281 MB             awk
1.625 MB                inet_gethost
2.52734 MB              ps
3.10547 MB              bash
3.12891 MB              /usr/bin/redis-server
3.30469 MB              postgres:
3.30469 MB              postgres:
3.71094 MB              bash
3.73047 MB              postgres:
4.14453 MB              nginx:
4.60156 MB              postgres:
4.94531 MB              nginx:
5.71875 MB              postgres:
14.5273 MB              /usr/bin/python
20.0703 MB              /usr/lib/postgresql/9.3/bin/postgres
22.2656 MB              statsd
26.4922 MB              node
37.5625 MB              node
37.6211 MB              node
40.5391 MB              /usr/bin/nodejs
45.1094 MB              /usr/bin/nodejs
45.3672 MB              node
49.5547 MB              /usr/bin/nodejs
97.168 MB               /usr/lib/erlang/erts-5.10.4/bin/beam.smp
404.887 MB              nginx:
405.301 MB              nginx:
405.352 MB              nginx:
405.367 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
405.371 MB              nginx:
root@69ccdaf9dcd5:/#
```

Any idea @agolybev?

**Comment by jkaberg** [comment]
By passing these environment variables to the container, I worked around the issue (now using ~300mb of ram).

```
      - NGINX_WORKER_PROCESSES=1
      - NGINX_WORKER_CONNECTIONS=1024
```

I'd still urge you to change the defaults tho.

**Comment by agolybev** [comment]
Hi @jkaberg,  thank you for report. 
By default nginx run one process per each processor core inside the container. I'll think about whether this is necessary.

**Comment by agolybev** [comment]
@jkaberg, we'll try testing one nginx worker per docker container by default: https://github.com/ONLYOFFICE/Docker-DocumentServer/commit/990a0ab127096105c6d5e5f57768e316f1bb3da6

**Comment by agolybev** [comment]
This is output from one of our production servers:
```
$ ps aux  | awk '{print $6/1024 " MB\t\t" $11}'  | sort -n | grep nginx
0.191406 MB		nginx:
0.28125 MB		nginx:
15.0977 MB		nginx:
15.1484 MB		nginx:
16.2422 MB		nginx:
16.3828 MB		nginx:
```
I really don't  know why the nginx inside your docker container uses to much memory.

**Comment by jkaberg** [comment]
@agolybev the output of the ```grep``` on my server results in 24 "cores" (12 core + 12 hyperthreads)

```
root@89c393a54433:/# grep processor /proc/cpuinfo | wc -l
24
```

That might give a hint to the high memory usage.

**Comment by agolybev** [comment]
Hi @jkaberg, I cann't push new image to Docker Hub right now. It will be released in Document Server v5.0.1 in the near future.

**Comment by Abijeet** [comment]
In case anyone is searching, I had  similar issue where the nginx process was taking up 22% of the 2GB RAM (450MB), 1 core server that I had. 

Setting these helped:

```
      - NGINX_WORKER_PROCESSES=1
      - NGINX_WORKER_CONNECTIONS=20
```

Which is fine in my case since the server will not be used by more than 3 people at the same time.
